packages/ui-docs-client/src/Header/index.js

```diff
@@ -39,14 +44,96 @@ import theme from './theme'
 class Header extends Component {
   static propTypes = {
     name: PropTypes.string.isRequired,
-    version: PropTypes.string
+    version: PropTypes.string,
+    versionsData: PropTypes.shape({
+      latestVersion: PropTypes.string.isRequired,
+      previousVersions: PropTypes.arrayOf(PropTypes.string).isRequired
+    })
   }
 
   static defaultProps = {
-    version: undefined
+    version: undefined,
+    versionsData: undefined
   }
 
+  handleSelect = (e, [selectedVersion]) => {
+    const { versionsData } = this.props
+    const { latestVersion } = versionsData
+    const isSelectedLatestVersion = selectedVersion === latestVersion
+    const splittedUrl = window.location.pathname.split('/').filter(Boolean)
+    const [versionInPath] = splittedUrl
+    const isOnLatestVersion = splittedUrl.length === 0
+
+    // 1: instructure.design, latest: v8, selected: v8 -> navigate to instructure.design/#index
+    // 2: instructure.design/v6/, latest: v8, selected: v6 -> navigate to instructure.design/v6/#index
+    if (
+      (isOnLatestVersion && isSelectedLatestVersion) ||
+      selectedVersion === versionInPath
+    ) {
+      return window.location.replace('#index')
+    }
+    // If we select the latest version from the dropdown,
+    // then navigate to the index (instructure.design/#currentHash).
+    // In every other case eg.: v6,v7 navigate to --> instructure.design/v6/#currentHash
+    const versionToNavigate = isSelectedLatestVersion
+      ? `/${window.location.hash}`
+      : `/${selectedVersion}/${window.location.hash}`
+
+    return window.location.replace(versionToNavigate)
+  }
+  renderVersionsBlock = () => {
+    const { versionsData } = this.props
+    const { latestVersion, previousVersions } = versionsData
+    const allVersions = [latestVersion, ...previousVersions]
+
+    const [versionInPath] = window.location.pathname.split('/').filter(Boolean)
+
+    const currentVersion = versionInPath || latestVersion
```

Comment:
Well, on localhost this comes back with v8, since "/v7" is not in the url. Will probably work in production, but probably not a bad idea to put a comment above it, so the devs know it's not a bug.

Reply:
Fixed

Reply:
I wouldn't hardcode "v8" here, soon it will be v9, and we won't remember fixing it in v7 :)